src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -86,11 +86,172 @@ private static IEnumerable<TextEdit> FormatRazor(FormattingContext context, Razo
                 // being overly careful to only try to format syntax forms they care about.
                 TryFormatCSharpCodeBlock(context, edits, source, node);
                 TryFormatSingleLineDirective(context, edits, source, node);
+                TryFormatTransitionCodeBlock(context, edits, source, node);
             }
 
             return edits;
         }
 
+        private static void TryFormatTransitionCodeBlock(FormattingContext context, IList<TextEdit> edits, RazorSourceDocument source, SyntaxNode node)
+        {
+            Range? openBraceRange = null;
+            Range? codeRange = null;
+            Range? closeBraceRange = null;
+            SyntaxNode? openBraceNode = null;
+            SyntaxNode? codeNode = null;
+            SyntaxNode? closeBraceNode = null;
+
+            // TODO: Split the if's into their own methods
+
+            // complex situations like
+            // @{
+            //  public void Method(){
+            //      @(DateTime.Now)
+            //  }
+            // }
+            if (node is CSharpRazorBlockSyntax csharpRazorBlock &&
+                csharpRazorBlock.Parent is CSharpCodeBlockSyntax firstCodeBlock &&
+                csharpRazorBlock.Parent.Parent is CSharpCodeBlockSyntax secondCodeBlock)
+            {
+                var csharpRazorRange = csharpRazorBlock.GetRange(source);
+                var childPosition = 0;
+                for (var i = 0; i < secondCodeBlock.Children.Count; i++)
```

Comment:
Another candidate for `IndexOf`? Maybe extension methods for `NextSibling` and `PreviousSibling` even?